src/main/ts/components/Chart.tsx

```diff
@@ -49,25 +53,53 @@ export default function Chart() {
     caycStartingDate,
     chartStartingDate,
     chartEndDate,
+    setSelectedProjects,
   } = useData();
 
-  if (isLoading) {
-    return <Spinner />;
-  }
+  const handleLoadProjects = async (q: string, resolve: (options: ProjectOption[]) => void) => {
+    const components = await getProjects({ q });
+    if (!components || components.length === 0) {
+      resolve([]);
+      return [];
+    }
+    const options = [
+      defaultProject,
+      ...components.map(({ key, name }) => {
+        return {
+          value: key,
+          label: name,
+        } as ProjectOption;
+      }),
+    ];
+    resolve(options);
+    return options;
+  };
+
+  const debouncedLoadProjects = React.useRef(
+    debounce((q: string, resolve: (options: ProjectOption[]) => void) => {
+      handleLoadProjects(q, resolve);
+    }, DEBOUNCE_DELAY),
+  );
+
+  const hasData =
+    cumulativeData?.length > 0 &&
+    caycProjectionData?.length > 0 &&
+    chartStartingDate &&
+    caycStartingDate &&
+    !isLoading;
```

Comment:
`isLoading` does not define whether or not there are data. Furthermore, it's already checked against on line 187.

```suggestion
  const hasData =
    cumulativeData?.length > 0 &&
    caycProjectionData?.length > 0 &&
    chartStartingDate &&
    caycStartingDate;
```